### PR TITLE
feat: add trivy scanner for code and images

### PR DESCRIPTION
**Pull request by JorTurFer**
Signed-off-by: Jorge Turrado <jorge_turrado@hotmail.es>

This PR enables [Trivy](https://aquasecurity.github.io/trivy/v0.29.2/) as an analyzer to check the code and the build images.
- On every PR Trivy will be executed checking the code and exposing the results as a table, in case of detected vulnerabilities, the CI will fail. 
- On every push to `main`, after the image generation, both images are analyzed for each arch to detect vulnerabilities and the results will be pushed to GitHub as security alerts

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
Relates to https://github.com/kedacore/keda/issues/1040

### CODE REVIEW

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
Please make sure to update our security policy as well

**Comment by pragmaticivan** [comment]
The `severity` field for this action defaults to `UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL`. 

Code LGTM although you might get some `extra noise` by not limiting that out for only `MEDIUM,HIGH,CRITICAL`.

**Comment by JorTurFer** [comment]
> The `severity` field for this action defaults to `UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL`. 
> 
> Code LGTM although you might get some `extra noise` by not limiting that out for only `MEDIUM,HIGH,CRITICAL`.

Yeah, I know that default severity includes lot of things but atm even that, we don't have any alert, so maybe it's better to have the most secure thing and just checking the noise. (but I agree also with modifying it)

File: SECURITY.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ We have a few preventive measures in place to detect security vulnerabilities:
   - Images are scanned in every pull request (PR) to detect new vulnerabilities.
   - Published images on GitHub Container Registry are monitored to detect new vulnerabilities so we can ship patches
 - [Whitesource Bolt for GitHub](https://www.whitesourcesoftware.com/free-developer-tools/bolt/) helps us with identifying vulnerabilities in our dependencies to raise awareness.
+- [Trivy](https://aquasecurity.github.io/trivy/latest/) helps us with identifying vulnerabilities in our dependencies and docker images to raise awareness.
```
Comment (tomkerkhove) [inline SECURITY.md:24]:
```suggestion
- [Trivy](https://aquasecurity.github.io/trivy/latest/) helps us with identifying vulnerabilities in our dependencies and docker images to raise awareness as part of our CI.
```

**Review by tomkerkhove — COMMENTED** [review]
This is great, but if we merge this then all PRs will be blocked because of the vulnerabilities.

Can you patch them as part of this PR please?

**Comment by JorTurFer** [comment]
> This is great, but if we merge this then all PRs will be blocked because of the vulnerabilities.
> 
> Can you patch them as part of this PR please?

Based on [lastest execution](https://github.com/kedacore/keda/runs/7116989844?check_suite_focus=true) there is not any vulnerability in the code

**Review by tomkerkhove — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]

**Comment by zroubalik** [comment]
One thing, don't we want to run this in nightly test runs as well? Or is that an overkill?

**Comment by JorTurFer** [comment]
> One thing, don't we want to run this in nightly test runs as well? Or is that an overkill?

good point, IDK because we do commits really often and this is executed on every merge to main, so maybe yes or maybe it's overkill, we could just add it, it's free xD

**Pull request merged**